### DISCUSSION TITLE
adGradSln and adGradSlnNeighbor

### DISCUSSION THREAD

**Discussion by heinono1** [idaholab/moose#25048] [Q&A General] [answered]
Hi. I need to evaluate gradients of two variables on element faces in an FV interface kernel code. AFAIK, var1().adSlnGrad(*_face_info,determineState()) will evaluate the gradient of var1 on the element face, but I am a little hazy on how to evaluate var2 on the same face when var2 lives in an element on the other side of the face. I have tried var2().adGradSln(_face_info->neighborPtr(),determineState()) but I am not sure this is correct (problems converging). There is also adGradSlnNeighbor but var2().adGradSlnNeighbor() does not work (it seems like adGradSlnNeighbor() does not want any arguments but all sorts of errors are generated by libMesh). Any comments, help, or suggestion are gratefully accepted.

Cheers,
Olle

**Comment by lindsayad** [comment, marked as answer]
For var2 have you tried `var2().adSlnGrad(*_face_info,determineState())`?

**Comment by heinono1** [comment]
Hi Alex. Yes, now I have. It  compiles but also gives convergence problems - to be honest, the convergence problems could be due to something else and not this particular issue.

Thanks,
Olle
